examples/nixnet/ethernet-frame-stream-input-reader.py

```diff
@@ -0,0 +1,183 @@
+        f"UTC Time:",
+        datetime.utcfromtimestamp(timestamp).strftime("%m/%d/%Y %H:%M:%S.%f")[:-3],
+        end=" ",
+    )
+
+
+# Declare all variables for the function
+i = 0
+count = 0
+session = None
+
+# Receive filter only applies for endpoint mode.
+flag = 3
+vid = 2
+priority = 3
+mac = "AA:BB:CC:DD:EE:FF"  # MAC Address filter is not enabled. Use random string.
+
+# Create the communication channel for the remote host and create connections to the NI-XNET and
+# session services.
+channel = grpc.insecure_channel(f"{SERVER_ADDRESS}:{SERVER_PORT}")
+client = grpc_nixnet.NiXnetStub(channel)
+
+print(CHOOSE_MONITOR_OR_ENDPOINT_TEXT)
+if sys.stdin.read(1) == "m":
+    INTERFACE = "ENET1/monitor"
+
+# Display parameters that will be used for the example.
+print(f"Interface: {INTERFACE}", end="\n")
+
+try:
+    # Create an XNET session in SignalOutSinglePoint mode
+    create_session_response = client.CreateSession(
+        nixnet_types.CreateSessionRequest(
+            database_name="",
+            cluster_name="",
+            list="",
+            interface=INTERFACE,
+            mode=nixnet_types.CREATE_SESSION_MODE_MODE_FRAME_IN_STREAM,
+        )
+    )
+    check_for_error(create_session_response.status)
+
+    session = create_session_response.session_ref
+    print("Session created successfully.\n")
+
+    rxfilter = nixnet_types.EptRxFilter(
+        use_flags=flag, vid=vid, priority=priority, destination_mac=mac
+    )
+
+    ept_rxfilter_array = nixnet_types.EptRxFilterArray(ept_rx_filter=[rxfilter])
+
+    # Set Receive Filter (Only applies for endpoint mode)
+    set_property_response = client.SetProperty(
+        nixnet_types.SetPropertyRequest(
+            session_ref=session,
+            property_id=nixnet_types.PROPERTY_SESSION_INTF_ENET_EPT_RECEIVE_FILTER,
+            ept_rx_filter_array=ept_rxfilter_array,
+        )
+    )
+    check_for_error(set_property_response.status)
+
+    print("Reading all received frames.\n")
+    print("Local Timestamp", "Network Timestamp", "Data")
+
+    while count < 10:
+        # Read the received frames into the buffer
+        read_frame_response = client.ReadFrame(
+            nixnet_types.ReadFrameRequest(
+                session_ref=session,
+                number_of_frames=NUM_OF_FRAMES,
+                max_payload_per_frame=MAX_PAYLOAD_PER_FRAME,
+                frame_type=nixnet_types.FRAME_TYPE_ENET,
+                timeout=nixnet_types.TIME_OUT_TIMEOUT_INFINITE,
+            )
+        )
+        check_for_error(read_frame_response.status)
+
+        frame_buffer = read_frame_response.buffer
+
+        for i in range(0, len(frame_buffer)):
+            print_timestamp(frame_buffer[i].enet.device_timestamp)
+            print_timestamp(frame_buffer[i].enet.network_timestamp)
+
+            for j in range(0, len(frame_buffer[i].enet.frame_data)):
+                print(
+                    hex(frame_buffer[i].enet.frame_data[j]).upper()[2:].zfill(2),
```

Comment:
Can you have a comment here regarding what hex() and zfill() functions do? overall what is the purpose of this entire line `hex(frame_buffer[i].enet.frame_data[j]).upper()[2:].zfill(2)`

Reply:
Added a comment.

Reply:
This isn't a helpful comment. What you are doing here is printing a 2-digit hexadecimal number without "0x" the prefix. Assuming that the input fits in 2 hex digits. You can also do that with format by the way:

```python
>>> hex(12)[2:].zfill(2)  
'0c'
>>> hex(23)[2:].zfill(2) 
'17'
>>> "{:02x}".format(12)   
'0c'
>>> "{:02x}".format(23) 
'17'
```

Reply:
Modified the same as you mentioned.